scipy/stats/mstats_basic.py

```diff
@@ -526,6 +526,47 @@ def _spearmanr_2cols(x):
         return SpearmanrResult(rs, prob)
 
 
+def _kendall_p_exact(n, c):
+    # Exact p-value, see Maurice G. Kendall, "Rank Correlation Methods" (4th Edition), Charles Griffin & Co., 1970.
+    if n <= 0:
+        raise ValueError
+    elif c < 0 or 4*c > n*(n-1):
+        raise ValueError
+    elif n == 1:
+        prob = 1.0
+    elif n == 2:
+        prob = 1.0
+    elif c == 0:
+        prob = 2.0/np.math.factorial(n)
+    elif c == 1:
+        prob = 2.0/np.math.factorial(n-1)
+    elif 4*c == n*(n-1):
+        prob = 1.0
+    elif n < 171:
+        new = np.zeros(c+1)
+        new[0] = 1.0
```

Comment:
`new[0:2] = 1.0`?  One line shorter and makes clear that the 1.0 is the same value twice, not 2 logically distinct values that just happen to be the same.

Reply:
OK, changed.